### ISSUE TITLE
hook-wx.lib.pubsub is not working.

### ISSUE THREAD

**Issue by sinancetinkaya** [pyinstaller/pyinstaller#2215]
```
D:\programlama\workspace\PRJ\TEST>"D:\Program Files\Python\Python27\Scripts\pyinstaller.exe" setup.spec --clean
131 INFO: PyInstaller: 3.3.dev0+b78bfe5
131 INFO: Python: 2.7.11
131 INFO: Platform: Windows-10-10.0.14393
131 INFO: UPX is not available.
131 INFO: Removing temporary files and cleaning cache in C:\Users\user\AppData\Roaming\pyinstaller
177 INFO: Extending PYTHONPATH with paths
['D:\\programlama\\workspace\\PRJ\\TEST',
 'D:\\programlama\\workspace\\PRJ\\TEST']
177 INFO: checking Analysis
177 INFO: Building Analysis because out00-Analysis.toc is non existent
177 INFO: Initializing module dependency graph...
177 INFO: Initializing module graph hooks...
230 INFO: running Analysis out00-Analysis.toc
230 INFO: Adding Microsoft.VC90.CRT to dependent assemblies of final executable
  required by d:\program files\python\python27\python.exe
299 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.1_none_8550c6b5d18a9128.manifest
299 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_f480bfaef65491a5.manifest
315 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_f47e3bf8f656f083.manifest
378 INFO: Searching for assembly x86_Microsoft.VC90.CRT_1fc8b3b9a1e18e3b_9.0.30729.9247_none ...
378 INFO: Found manifest C:\WINDOWS\WinSxS\Manifests\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_5090cb78bcba4a35.manifest
378 INFO: Searching for file msvcr90.dll
378 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_5090cb78bcba4a35\msvcr90.dll
378 INFO: Searching for file msvcp90.dll
378 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_5090cb78bcba4a35\msvcp90.dll
378 INFO: Searching for file msvcm90.dll
378 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_5090cb78bcba4a35\msvcm90.dll
447 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.1_none_8550c6b5d18a9128.manifest
447 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_f480bfaef65491a5.manifest
447 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_f47e3bf8f656f083.manifest
447 INFO: Adding redirect Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
532 INFO: Caching module hooks...
532 INFO: Analyzing main.py
4306 INFO: Processing pre-safe import module hook   _xmlplus
4454 INFO: Processing pre-find module path hook   distutils
11838 INFO: Loading module hooks...
11838 INFO: Loading module hook "hook-distutils.py"...
11838 INFO: Loading module hook "hook-sysconfig.py"...
11838 INFO: Loading module hook "hook-xml.py"...
11838 INFO: Loading module hook "hook-lxml.etree.py"...
11852 INFO: Loading module hook "hook-httplib.py"...
11852 INFO: Loading module hook "hook-_tkinter.py"...
12037 INFO: checking Tree
12037 INFO: Building Tree because out00-Tree.toc is non existent
12037 INFO: Building Tree out00-Tree.toc
12216 INFO: checking Tree
12216 INFO: Building Tree because out01-Tree.toc is non existent
12216 INFO: Building Tree out01-Tree.toc
12237 INFO: Loading module hook "hook-selenium.py"...
12237 INFO: Loading module hook "hook-xml.dom.domreg.py"...
12237 INFO: Loading module hook "hook-boto.py"...
12269 INFO: Loading module hook "hook-pkg_resources.py"...
12629 INFO: Processing pre-safe import module hook   win32com
12944 INFO: Loading module hook "hook-requests.py"...
12950 INFO: Loading module hook "hook-certifi.py"...
12953 INFO: Loading module hook "hook-pywintypes.py"...
13239 INFO: Loading module hook "hook-keyring.backends.py"...
13461 INFO: Loading module hook "hook-encodings.py"...
13940 INFO: Loading module hook "hook-sqlite3.py"...
14024 INFO: Loading module hook "hook-wx.lib.pubsub.py"...
Traceback (most recent call last):
  File "<string>", line 41, in <module>
  File "<string>", line 36, in walk_packages
  File "<string>", line 20, in walk_packages
  File "d:\program files\python\python27\lib\site-packages\wx-3.0-msw\wx\lib\pubsub\core\arg1\__init__.py", line 16, in <module>
    raise RuntimeError(msg)
RuntimeError: Should not import this directly, used by pubsub.core if applicable
14394 WARNING: Hidden import "wx.lib.pubsub.core.publisher" not found!
14394 WARNING: Hidden import "wx.lib.pubsub.core.listenerimpl" not found!
14394 WARNING: Hidden import "wx.lib.pubsub.core.publishermixin" not found!
14394 WARNING: Hidden import "wx.lib.pubsub.core.datamsg" not found!
14394 WARNING: Hidden import "wx.lib.pubsub.core.topicargspecimpl" not found!
14394 WARNING: Hidden import "wx.lib.pubsub.core.topicmgrimpl" not found!
14394 INFO: Loading module hook "hook-win32com.py"...
14812 INFO: Loading module hook "hook-pythoncom.py"...
15160 INFO: checking Tree
15160 INFO: Building Tree because out02-Tree.toc is non existent
15160 INFO: Building Tree out02-Tree.toc
15165 INFO: Looking for ctypes DLLs
15181 INFO: Analyzing run-time hooks ...
15197 INFO: Including run-time hook 'pyi_rth__tkinter.py'
15197 INFO: Including run-time hook 'pyi_rth_pkgres.py'
15197 INFO: Including run-time hook 'pyi_rth_win32comgenpy.py'
15213 INFO: Looking for dynamic libraries
15481 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.1_none_7dd1e0ebd6590e0b.manifest
15481 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_ed01d9e4fb230e88.manifest
15545 INFO: Searching for assembly x86_Microsoft.VC90.MFC_1fc8b3b9a1e18e3b_9.0.30729.6161_none ...
15545 INFO: Found manifest C:\WINDOWS\WinSxS\Manifests\x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_4bf7e3e2bf9ada4c.manifest
15545 INFO: Searching for file mfc90.dll
15545 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_4bf7e3e2bf9ada4c\mfc90.dll
15545 INFO: Searching for file mfc90u.dll
15545 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_4bf7e3e2bf9ada4c\mfc90u.dll
15545 INFO: Searching for file mfcm90.dll
15545 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_4bf7e3e2bf9ada4c\mfcm90.dll
15545 INFO: Searching for file mfcm90u.dll
15545 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_4bf7e3e2bf9ada4c\mfcm90u.dll
15611 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.1_none_7dd1e0ebd6590e0b.manifest
15611 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_ed01d9e4fb230e88.manifest
15611 INFO: Adding redirect Microsoft.VC90.MFC version (9, 0, 21022, 8) -> (9, 0, 30729, 6161)
17603 INFO: Looking for eggs
17603 INFO: Using Python library d:\program files\python\python27\python27.dll
17603 INFO: Found binding redirects:
[BindingRedirect(name=u'Microsoft.VC90.CRT', language=None, arch=u'x86', oldVersion=(9, 0, 21022, 8), newVersion=(9, 0, 30729, 9247), publicKeyToken=u'1fc8b3b9a1e18e3b')
, BindingRedirect(name=u'Microsoft.VC90.MFC', language=None, arch=u'x86', oldVersion=(9, 0, 21022, 8), newVersion=(9, 0, 30729, 6161), publicKeyToken=u'1fc8b3b9a1e18e3b'
)]
17603 INFO: Warnings written to D:\programlama\workspace\PRJ\TEST\build\setup\warnsetup.txt
18019 INFO: checking PYZ
18020 INFO: Building PYZ because out00-PYZ.toc is non existent
18020 INFO: Building PYZ (ZlibArchive) D:\programlama\workspace\PRJ\TEST\build\setup\out00-PYZ.pyz
20273 INFO: checking PKG
20273 INFO: Building PKG because out00-PKG.toc is non existent
20273 INFO: Building PKG (CArchive) out00-PKG.pkg
20312 INFO: Bootloader d:\program files\python\python27\lib\site-packages\PyInstaller\bootloader\Windows-32bit\runw.exe
20312 INFO: checking EXE
20312 INFO: Building EXE because out00-EXE.toc is non existent
20312 INFO: Building EXE from out00-EXE.toc
20328 INFO: SRCPATH [('.\\resources\\app.ico', None)]
20328 INFO: Updating icons from ['.\\resources\\app.ico'] to e:\temp\tmp28x64j
20328 INFO: Writing RT_GROUP_ICON 0 resource with 20 bytes
20328 INFO: Writing RT_ICON 1 resource with 51240 bytes
20328 INFO: Appending archive to EXE D:\programlama\workspace\PRJ\TEST\build\setup\TEST.exe
20358 INFO: checking COLLECT
20358 INFO: Building COLLECT because out00-COLLECT.toc is non existent
WARNING: The output directory "D:\programlama\workspace\PRJ\TEST\dist\TEST_3.00.00" and ALL ITS CONTENTS will be REMOVED! Continue? (y/n)y
23027 INFO: Removing dir D:\programlama\workspace\PRJ\TEST\dist\TEST_3.00.00
23499 INFO: Building COLLECT out00-COLLECT.toc
23513 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23568 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23568 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\python27.dll
23568 INFO: Updating resource type 24 name 2 language 1033
23628 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23628 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\_hashlib.pyd
23628 INFO: Updating resource type 24 name 2 language 1033
23628 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23644 INFO: Redirecting Microsoft.VC90.MFC version (9, 0, 21022, 8) -> (9, 0, 30729, 6161)
23644 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\win32ui.pyd
23644 INFO: Updating resource type 24 name 2 language 1033
23658 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23658 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\bz2.pyd
23658 INFO: Updating resource type 24 name 2 language 1033
23658 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23658 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\_ctypes.pyd
23674 INFO: Updating resource type 24 name 2 language 1033
23674 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23674 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\_ssl.pyd
23674 INFO: Updating resource type 24 name 2 language 1033
23697 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23697 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\_socket.pyd
23697 INFO: Updating resource type 24 name 2 language 1033
23713 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23713 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\pyexpat.pyd
23713 INFO: Updating resource type 24 name 2 language 1033
23713 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23713 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\select.pyd
23713 INFO: Updating resource type 24 name 2 language 1033
23727 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23727 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\unicodedata.pyd
23727 INFO: Updating resource type 24 name 2 language 1033
23743 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23743 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\_tkinter.pyd
23743 INFO: Updating resource type 24 name 2 language 1033
23759 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23759 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\_bsddb.pyd
23759 INFO: Updating resource type 24 name 2 language 1033
23774 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23774 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\_testcapi.pyd
23774 INFO: Updating resource type 24 name 2 language 1033
23796 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\lxml.etree.pyd
23796 INFO: Updating resource type 24 name 2 language 1033
23828 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23828 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\_multiprocessing.pyd
23828 INFO: Updating resource type 24 name 2 language 1033
23828 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
23828 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\winsound.pyd
23828 INFO: Updating resource type 24 name 2 language 1033
23843 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\wx._misc_.pyd
23843 INFO: Updating resource type 24 name 2 language 1033
23858 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\wx._controls_.pyd
23858 INFO: Updating resource type 24 name 2 language 1033
23875 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\wx._windows_.pyd
23875 INFO: Updating resource type 24 name 2 language 1033
23897 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\wx._gdi_.pyd
23897 INFO: Updating resource type 24 name 2 language 1033
24013 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\wx._core_.pyd
24013 INFO: Updating resource type 24 name 2 language 1033
24013 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\wx._xrc.pyd
24013 INFO: Updating resource type 24 name 2 language 1033
24029 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\wx._grid.pyd
24029 INFO: Updating resource type 24 name 2 language 1033
24045 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
24045 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\_sqlite3.pyd
24045 INFO: Updating resource type 24 name 2 language 1033
24092 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\mfc90.dll
24098 INFO: Updating resource type 24 name 1000 language 1033
24161 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\mfc90u.dll
24175 INFO: Updating resource type 24 name 1000 language 1033
24244 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
24244 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\tcl85.dll
24244 INFO: Updating resource type 24 name 2 language 1033
24276 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\tk85.dll
24276 INFO: Updating resource type 24 name 1 language 1033
24412 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
24428 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\tk85.dll
24444 INFO: Updating resource type 24 name 2 language 1033
24645 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
24645 INFO: Updating manifest in C:\Users\user\AppData\Roaming\pyinstaller\bincache00_py27_32bit\sqlite3.dll
24645 INFO: Updating resource type 24 name 2 language 1033

```

**Comment by dhyams** [comment]
Could you attach an example script that reproduces the problem?  I don't have trouble with this hook; works OK for me on three platforms.

**Comment by sinancetinkaya** [comment]
Ok, I just wrote a test script to reproduce this error.

log.py

```
import threading
import time
import wx
from wx.lib.pubsub import pub


class MyDialog(wx.Dialog):
    def __init__(self):
        wx.Dialog.__init__(self, None)
        self.textCtrl = wx.TextCtrl(self, wx.ID_ANY, wx.EmptyString, wx.DefaultPosition, wx.DefaultSize,
                                      wx.HSCROLL | wx.TE_MULTILINE | wx.TE_READONLY | wx.TE_WORDWRAP)
        bSizer = wx.BoxSizer(wx.VERTICAL)
        bSizer.Add(self.textCtrl, 1, wx.EXPAND, 5)
        self.SetSizer(bSizer)
        self.Layout()
        pub.subscribe(self.log, 'log')
        self.Bind(wx.EVT_CLOSE, self.onClose)

    def log(self, msg):
        self.textCtrl.AppendText(msg + '\n')
        self.textCtrl.ScrollPages(1)
        pass

    def onClose(self, event):
        self.Destroy()
        event.Skip()


class Thread(threading.Thread):
    def __init__(self):
        threading.Thread.__init__(self)

    def run(self):
        for i in range(10):
            pub.sendMessage("log", msg="This is the {}. test message.".format(i))
            time.sleep(1)


app = wx.App()
dialog = MyDialog()
t = Thread()
t.daemon = True
t.start()
dialog.ShowModal()

app.MainLoop()

```

log.spec

```
# -*- mode: python -*-
import os, sys

a = Analysis(["log.py"],
             pathex=[],
             hiddenimports=[],
             hookspath=[],
             runtime_hooks=None,
             excludes=None,
             cipher=None)

pyz = PYZ(a.pure, cipher=None)

exe = EXE(pyz,
          a.scripts,
          exclude_binaries=True,
          name=os.path.join('dist', 'log.exe'),
          debug=False,
          strip=None,
          upx=True,
          console=False,
          icon=None)

coll = COLLECT(exe,
               a.binaries,
               a.zipfiles,
               a.datas,
               strip=None,
               upx=True,
               name="log")

```
And here is the build log:
```
141 INFO: PyInstaller: 3.3.dev0+b78bfe5
141 INFO: Python: 2.7.11
141 INFO: Platform: Windows-10-10.0.14393
142 INFO: UPX is not available.
145 INFO: Extending PYTHONPATH with paths
['D:\\programlama\\workspace\\user\\TEST']
145 INFO: checking Analysis
145 INFO: Building Analysis because out00-Analysis.toc is non existent
145 INFO: Initializing module dependency graph...
150 INFO: Initializing module graph hooks...
201 INFO: running Analysis out00-Analysis.toc
206 INFO: Adding Microsoft.VC90.CRT to dependent assemblies of final executable
  required by d:\program files\python\python27\python.exe
279 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.1_none_8550c6b5d18a9128.manifest
280 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_f480bfaef65491a5.manifest
282 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_f47e3bf8f656f083.manifest
354 INFO: Searching for assembly x86_Microsoft.VC90.CRT_1fc8b3b9a1e18e3b_9.0.30729.9247_none ...
354 INFO: Found manifest C:\WINDOWS\WinSxS\Manifests\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_5090cb78bcba4a35.manifest
355 INFO: Searching for file msvcr90.dll
355 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_5090cb78bcba4a35\msvcr90.dll
355 INFO: Searching for file msvcp90.dll
355 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_5090cb78bcba4a35\msvcp90.dll
355 INFO: Searching for file msvcm90.dll
355 INFO: Found file C:\WINDOWS\WinSxS\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_5090cb78bcba4a35\msvcm90.dll
430 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.1_none_8550c6b5d18a9128.manifest
431 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_f480bfaef65491a5.manifest
433 INFO: Found C:\WINDOWS\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_f47e3bf8f656f083.manifest
434 INFO: Adding redirect Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)
517 INFO: Caching module hooks...
522 INFO: Analyzing log.py
4144 INFO: Processing pre-safe import module hook   _xmlplus
4276 INFO: Processing pre-find module path hook   distutils
5029 INFO: Loading module hooks...
5029 INFO: Loading module hook "hook-distutils.py"...
5030 INFO: Loading module hook "hook-sysconfig.py"...
5030 INFO: Loading module hook "hook-xml.py"...
5095 INFO: Loading module hook "hook-httplib.py"...
5096 INFO: Loading module hook "hook-_tkinter.py"...
5272 INFO: checking Tree
5272 INFO: Building Tree because out00-Tree.toc is non existent
5272 INFO: Building Tree out00-Tree.toc
5453 INFO: checking Tree
5453 INFO: Building Tree because out01-Tree.toc is non existent
5453 INFO: Building Tree out01-Tree.toc
5474 INFO: Loading module hook "hook-encodings.py"...
5946 INFO: Loading module hook "hook-wx.lib.pubsub.py"...
Traceback (most recent call last):
  File "<string>", line 41, in <module>
  File "<string>", line 36, in walk_packages
  File "<string>", line 20, in walk_packages
  File "d:\program files\python\python27\lib\site-packages\wx-3.0-msw\wx\lib\pubsub\core\arg1\__init__.py", line 16, in <module>
    raise RuntimeError(msg)
RuntimeError: Should not import this directly, used by pubsub.core if applicable
6300 WARNING: Hidden import "wx.lib.pubsub.core.publisher" not found!
6302 WARNING: Hidden import "wx.lib.pubsub.core.listenerimpl" not found!
6302 WARNING: Hidden import "wx.lib.pubsub.core.publishermixin" not found!
6302 WARNING: Hidden import "wx.lib.pubsub.core.datamsg" not found!
6303 WARNING: Hidden import "wx.lib.pubsub.core.topicargspecimpl" not found!
6305 WARNING: Hidden import "wx.lib.pubsub.core.topicmgrimpl" not found!
6323 INFO: checking Tree
6323 INFO: Building Tree because out02-Tree.toc is non existent
6323 INFO: Building Tree out02-Tree.toc
6355 INFO: Looking for ctypes DLLs
6368 INFO: Analyzing run-time hooks ...
6371 INFO: Including run-time hook 'pyi_rth__tkinter.py'
6377 INFO: Looking for dynamic libraries
7744 INFO: Looking for eggs
7744 INFO: Using Python library d:\program files\python\python27\python27.dll
7744 INFO: Found binding redirects:
[BindingRedirect(name=u'Microsoft.VC90.CRT', language=None, arch=u'x86', oldVersion=(9, 0, 21022, 8), newVersion=(9, 0, 30729, 9247), publicKeyToken=u'1fc8b3b9a1e
18e3b')]
7747 INFO: Warnings written to D:\programlama\workspace\user\TEST\build\log\warnlog.txt
8009 INFO: checking PYZ
8009 INFO: Building PYZ because out00-PYZ.toc is non existent
8009 INFO: Building PYZ (ZlibArchive) D:\programlama\workspace\user\TEST\build\log\out00-PYZ.pyz
9040 INFO: checking PKG
9040 INFO: Building PKG because out00-PKG.toc is non existent
9042 INFO: Building PKG (CArchive) out00-PKG.pkg
9076 INFO: Bootloader d:\program files\python\python27\lib\site-packages\PyInstaller\bootloader\Windows-32bit\runw.exe
9076 INFO: checking EXE
9078 INFO: Building EXE because out00-EXE.toc is non existent
9078 INFO: Building EXE from out00-EXE.toc
9078 INFO: Appending archive to EXE D:\programlama\workspace\user\TEST\build\log\log.exe
9092 INFO: checking COLLECT
9092 INFO: Building COLLECT because out00-COLLECT.toc is non existent
9092 INFO: Building COLLECT out00-COLLECT.toc
9111 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 9247)


```

**Comment by dhyams** [comment]
I can confirm this behavior in the development branch.  PyInstaller 3.2 does not have this problem, so the problem is at least localized to something that has been changed in the current "develop" branch.

**Comment by dhyams** [comment]
I run a git bisect to investigate this; the commit pointed-to as being the one that causes this problem is 4748b0d77

**Comment by dhyams** [comment]
Maybe @bjones1 could comment?  I see that this commit is part of a larger effort to fix several submodule bugs, but I'm not sure of the implications/advantages of using pkgutil rather than the code that was already there.

At any rate, hopefully this will not be hard to fix, since we have a good before/after point.

**Comment by sinancetinkaya** [comment]
I think the problem has an explanation in the __init__.py that pyinstaller is trying to import.
d:\program files\python\python27\lib\site-packages\wx-3.0-msw\wx\lib\pubsub\core\arg1\__init__.py:
```
"""
This is not really a package init file, it is only here to simplify the
packaging and installation of pubsub.core's protocol-specific subfolders
by setuptools.  The python modules in this folder are automatically made
part of pubsub.core via pubsub.core's __path__. Hence, this should not
be imported directly, it is part of pubsub.core when the messaging
protocol is "arg1" (and not usable otherwise).

:copyright: Copyright since 2006 by Oliver Schoenborn, all rights reserved.
:license: BSD, see LICENSE_BSD_Simple.txt for details.

"""


msg = 'Should not import this directly, used by pubsub.core if applicable'
raise RuntimeError(msg)
```
As you can see __init__.py itself raises this error because pyinstaller imports __init__.py directly.

**Comment by dhyams** [comment]
That's true, sorry that I left that unsaid above.  I was reading the docs for pkgutil a while ago, and it attempts to import things, which the original code (that this commit replaces) does not.  Since it tries to import, this error is triggered.

So the larger question is how to make the new code work properly (since it solves some other pyinstaller issues) with modules, such as wx.lib.pubsub.core.arg1, that throw exceptions if an import of them is attempted. 

An interesting experiment would be to just remove the "raise RuntimeError" from the __init__.py referred to above, and see if pyinstaller finished out properly and packages up the wx.lib.pubsub submodules properly.

**Comment by sinancetinkaya** [comment]
My original code is much larger than the test code and it was not working even if I remove the "raise RuntimeError" from the `__init__.py`
But I was using the development build of pyinstaller.
After the frustration of this bug, I changed the design of the script and ditched pubsub.

**Comment by dhyams** [comment]
Did you remove that from both arg1/__init__.py, and kwargs/__init__.py.  You have to make sure to get both of them.

If you had used version 3.2 of Pyinstaller (the latest), this wouldn't have happened...the commit in question happened between the release of 3.2 and now.  Thanks so much for detecting the problem; this way it can be fixed before the next version of Pyinstaller is release, even though I'm not very sure as to what to do to fix it.

**Comment by htgoebel** [comment]
FYI: The way wx-pubsub imports the concrete implementation is very, very tricky, not to say hackish. Solving this already took a lot of time and obviously we did not manage to get it right. I'm sorry for the trouble you had with this and am going to add a "Known Bugs" section to the documentation.

**Comment by dhyams** [comment]
@htgoebel: absolutely, it isn't pretty.  My only point above is that it worked in Pyinstaller 3.2, but not in the current dev branch, and there is a small bit of code that is causing the problem (albeit a significant change in how submodules are found).

Failing a way to fix it "right", maybe we can wrap the existing code in a try/except that catches RuntimeError, and revert back to the old ways of doing things if one is caught.  Not pleasing, but in my mind better than leaving it as a known bug.

**Comment by sinancetinkaya** [comment]
@htgoebel @dhyams Thank you guys for all the work you have done for this project!

**Comment by bjones1** [comment]
@dhyams, unfortunately I'm busy with other projects and don't have the time to really investigate this. I'd be happy to review a pull request -- it sounds like catching an exception may help?

**Comment by dhyams** [comment]
@bjones1 no worries; I thought that maybe you might have some comments concerning the motivation behind the change...didn't want you necessarily to fix it!

**Comment by bjones1** [comment]
See #2007 - this version supports .eggs.

**Comment by dhyams** [comment]
Fixed in pull request #2233.

**Comment by giumas** [comment]
The pull request https://github.com/pyinstaller/pyinstaller/pull/2233 works with wx Phoenix on Win10 and Python 3.6.
